src/python/pants/engine/rules.py

```diff
@@ -69,7 +69,7 @@ def visit_Call(self, node: ast.Call) -> None:
 
 
 @memoized
-def subsystem_rule(optionable_factory: Type[OptionableFactory]) -> "TaskRule":
+def SubsystemRule(optionable_factory: Type[OptionableFactory]) -> "TaskRule":
```

Comment:
This now violates Python naming conventions. You should always use snake_case for function names. But, I argue that we should optimize for the hundreds of call sites we'll have, rather than purity here.

We could refactor this to be a class with `__call__` defined. I didn't do that because I don't know how to preserve the `@memoized` semantics.

Reply:
Might as well document that fact here, so the reader knows that the author is intentionally violating style, and why.